mtsmorf/io/move/read.py

```
@@ -0,0 +1,561 @@
+    end_trial_id: str = None
+    trial_duration: float = None
+    avg_speed: float = None
+    std_speed: float = None
+
+    def __repr__(self):
+        """Taken from mne-python."""
+        MAX_WIDTH = 68
+        strs = [f"<{self.subject} Move Trial {self.trial_idx} | %s non-empty values"]
+        non_empty = 0
+        for k, v in self.__dict__.items():
+            if k == "ch_names":
+                if v:
+                    entr = shorten(", ".join(v), MAX_WIDTH, placeholder=" ...")
+                else:
+                    entr = "[]"  # always show
+                    non_empty -= 1  # don't count as non-empty
+            elif k in ['index', 'subject']:
+                continue
+            elif k in ["speed_instruction"]:
+                entr = v
+            else:
+                try:
+                    this_len = len(v)
+                except TypeError:
+                    entr = "{}".format(v) if v is not None else ""
+                else:
+                    if this_len > 0:
+                        entr = "%d item%s (%s)" % (
+                            this_len,
+                            _pl(this_len),
+                            type(v).__name__,
+                        )
+                    else:
+                        entr = ""
+            if entr != "":
+                non_empty += 1
+                strs.append("%s: %s" % (k, entr))
+        st = "\n ".join(sorted(strs))
+        st += "\n>"
+        st %= non_empty
+        return st
+
+    def to_data_frame(self):
+        df = pd.DataFrame.from_dict(self.__dict__.items(), orient='columns').T
+        df.columns = df.iloc[0]
+
+        # set datatype
+        df['beh_fname'] = df['beh_fname'].astype(str)
+        df['event_fname'] = df['event_fname'].astype(str)
+
+        df.drop(df.index[0], inplace=True)
+        return df
+
+    def _add_xy_metrics(self, metrics_df):
+        if not self.success or self.perturbed:
+            return
```

would return a warning instead of failing silently:

`warnings.warn("xy metrics not added because Trial instance includes unsuccessful or perturbed trials.)`

just `import warnings` if you include this